src/main/java/it/gov/pagopa/pu/registry/mapper/installment/DebtPositionIoEventDTO2InstallmentRegistryMapper.java

```diff
@@ -0,0 +1,33 @@
+package it.gov.pagopa.pu.registry.mapper.installment;
+
+import it.gov.pagopa.pu.registry.event.payments.dto.DebtPositionIoEventDTO;
+import it.gov.pagopa.pu.registry.model.InstallmentRegistry;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class DebtPositionIoEventDTO2InstallmentRegistryMapper {
+
+  public List<InstallmentRegistry> map(DebtPositionIoEventDTO dto) {
+    if (dto.getPayload().getMessages() == null) return List.of();
+
+    return dto.getPayload().getMessages().stream()
+      .filter(ioMessageDTO -> ioMessageDTO.getNav() != null && !ioMessageDTO.getNav().isEmpty())
+      .map(ioMessageDTO -> InstallmentRegistry.builder()
+        .eventId(dto.getEventId() + "." + ioMessageDTO.getNav())
+        .eventType(dto.getEventType())
+        .traceId(dto.getTraceId())
+        .eventDateTime(dto.getEventDateTime())
+        .eventDescription(dto.getEventDescription())
+        .debtPositionId(dto.getPayload().getDebtPositionId())
+        .organizationId(dto.getPayload().getOrganizationId())
+        .nav(ioMessageDTO.getNav())
+        .build())
+      .collect(Collectors.toList());
```

Comment:
Your IDE doesn't signal this as a warning?
```suggestion
      .toList();
```

Reply:
No.